File: android/app/src/main/kotlin/org/iespik/printer/PrintResult.kt
package org.iespik.printer

data class PrintResult(
    val success: Boolean,
    val message: String
) {
    companion object {
        val InvalidPathError: PrintResult = PrintResult(false, "Invalid File Path")
        val PrinterNotFoundError: PrintResult = PrintResult(false, "Printer Not Found")
        val Success: PrintResult = PrintResult(true, "Success")
    }

    fun toMap(): Map<String, Any> {
        return mapOf<String, Any>("success" to success, "message" to message)
    }
}

File: lib/core/domains/printer/entities/print_result.dart
class PrintResult {
  final bool success;
  final String? message;

  PrintResult(this.success, this.message);

  PrintResult.failed(String message) : this(false, message);

  factory PrintResult.fromJson(Map<dynamic, dynamic> map) {
    return PrintResult(map["success"] as bool, map["message"] as String?);
  }
}
